airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -97,7 +97,9 @@ def backoff_time(self, response: requests.Response) -> Optional[float]:
          :return how long to backoff in seconds. The return value may be a floating point number for subsecond precision. Returning None defers backoff
          to the default backoff behavior (e.g using an exponential algorithm).
         """
-        return self._requester.backoff_time(response)
+        should_retry = self._requester.should_retry(response)
+        assert isinstance(should_retry, RetryResponseStatus)
```

Comment:
i'm not convinced we want an assert, but it would be unexpected to receive anything else here because the HttpStream always checks if the request should be retried before calling backoff_time https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L291-L292